Updating tests for Android changes

burtonemily opened the pull request:
- Removed a bunch of control message checks 
- Update system workflows (adding media, allow notifications)
- Extracting workflows for sending media to own functions (sendDocument, sendVideo, etc)
- Starting to add linked device checks to all tests (and removing them as separate tests)

burtonemily committed:
Finishes message checks for ios testing images, videos, voice messages, links, GIFs and long messages. Also adds test to check deletion functionality in groups

burtonemily committed:
Adding files that were untracked for group message checks

burtonemily committed:
Start of disappearing messages tests

burtonemily committed:
Starting disappear after send test

burtonemily committed:
working state with yarn import

burtonemily committed:
fix: move to yarn working!

Bilb committed:
Merge pull request #6 from burtonemily/move-to-yarn-try-2

Move to yarn try 2

burtonemily committed:
Fix: deleteText function to combine ios and android functionality

burtonemily committed:
Attempting to fix longPress on username tests

burtonemily committed:
Adds in disappearing messages tests, creates a new type for conversationType and DMTimeOption, extracts set disappearing messages into a separate function and starts disappearing messages checks for different types of messages

burtonemily committed:
Forgot to add untracked files

burtonemily committed:
Adds in getAttribute function to validate selected radio button

burtonemily committed:
Adds in getAttribute function into disappearing messages tests to verify default time is selected

burtonemily committed:
Replaced function waitForElementToBePresent with new function waitForTextElementToBePresent, which incorporates a loop to continually check instead of static wait

burtonemily committed:
extending timeout for deleted message and commenting out send doc test until fixed

burtonemily committed:
Disappearing messages changes

burtonemily committed:
Adding in tests and types for call history deletion and community invites

burtonemily committed:
fixing typo

burtonemily committed:
feat: disappear messages

burtonemily committed:
Fixed merge conflicts

burtonemily committed:
Found more merge conflicts

burtonemily committed:
Adding to tests to configure to android, updated some aaccessibility id types and updated android version

burtonemily committed:
Fixing tests for android updates

burtonemily committed:
Fixing merge conflicts

burtonemily committed:
Changing Message Body to Message body and replacing all waitForElementToBePresent functions with waitForTextElementToBePresent

burtonemily committed:
Changed wording of tests and updated sleepFor timings

burtonemily committed:
Merging dev into main: includes disappearing messages tests, group message checks.

burtonemily committed:
Deletes separate test mentions for ios and android, converts createGroup into an object, updates change username test, deletes unused functions. Still working through feedback

burtonemily committed:
Fixing tests based on android capabilities, refactored some code

burtonemily committed:
Cleaning up sleepFor and replacing loading animation function

burtonemily committed:
Adds in new type for XY coordinates, interaction points, removes sleepFor in various locations, based on feedback, adds in disappearing messages tests to main repo (removes test skips), and updates image addition for profile picture test

burtonemily committed:
Commenting out sendDoc because pdf file sending isnt working on ios, also fixes some minor quotation changes

burtonemily committed:
Adding in a joinCommunity function because its used in multiple places and changing some accessibility types

burtonemily committed:
Adds joinCommunity

burtonemily committed:
Finishes off items from review, fixes setNickname ios test and other stuff

burtonemily committed:
Updating tests based on disappearing messages changes

burtonemily committed:
Added a new function waitForControlMessageToBePresent to replace findConfig and to centralise all functions into one

burtonemily committed:
Cleaning up test messages and combining logic into sendimage function

burtonemily committed:
Updates typescript and appium versions to be compatible with xcuitest for xcode and ios testing

burtonemily committed:
Updates yarn and removes npm

Bilb committed:
fix: update to yarn 4

Bilb committed:
Merge remote-tracking branch 'emily/dev' into update-yarn-4

burtonemily committed:
Merge pull request #1 from Bilb/update-yarn-4

fix: update to yarn 4

burtonemily committed:
fix: tests and inspector are working

burtonemily committed:
fix typings for uiautomator

burtonemily committed:
Updating control messages based on new copy and starting a function to dynamically figure out control message text

burtonemily committed:
Working on setDisappearingMessages function

burtonemily committed:
working on control message abstraction

burtonemily committed:
Adding in follow settings functionality to disappearing messages tests

burtonemily committed:
Typifies control messages

burtonemily committed:
Adding more tests for community checks

burtonemily committed:
Hardcoded usernames and group names as types

burtonemily committed:
Fixes gestures error ios

burtonemily committed:
Merge branch 'dev' into disappearing_messages_test

burtonemily committed:
Merged dev into branch

burtonemily committed:
Updating accessibility ids to match android

burtonemily committed:
Fixes send image test to update to new os

burtonemily committed:
Refactoring some sent status issues

burtonemily committed:
Extending timeout for message request banner, its taking longer than usual

burtonemily committed:
Changes 1o1 to 1:1, updates accessibility ids that changed with ios update, minor refactor on tests

burtonemily committed:
Fixing merge conflicts

burtonemily committed:
Last addition to disappearing message changes

burtonemily committed:
Updates interaction points, test file, send doc test for ios and fixes voice messages teset

burtonemily committed:
Merging dev into onboarding

burtonemily committed:
Changed session ID to account ID, removed platform from newContact and updated recovery phrase to be recovery phrase

burtonemily committed:
feat: Enhance test scripts and update element selectors

- Add a new script `test-grep` to run tests based on a specific grep pattern, allowing for more targeted test runs.
- Update element selectors in test scripts for improved reliability and compatibility with the latest app version.
- Implement retries for clicking on "Untrusted attachment message" until "Download media" appears, improving test robustness.
- Adjust sleep durations and add explicit waits to ensure elements are interactable before proceeding, reducing flakiness in tests.
- Update Android app path to version 1.18.1-arm64-v8a, ensuring tests run against the latest build.
- Enable `reduceMotion` in iOS capabilities to improve test speed and consistency.
- Log the full path of the Android app being tested for better debugging and traceability.
- Refactor `installAppToDeviceName` utility function to include timing logs for each step, aiding in performance analysis.
- Implement error handling for stale element references, adding a retry mechanism for clicks.
- Modify `clickOnElementAll` to return the clicked element, allowing for further actions or checks.
- Increase long press duration in `longClick` method to ensure the action is recognized.
- Add `maxWait` parameter to loading animation wait function, providing better control over wait times.
- Update `ControlMessage` and `AccessibilityId` types to include new scenarios and elements, reflecting app changes.
- Rename describe blocks for clarity and consistency across test suites.feat: Update newly inputted data-testids to make tests more stable

burtonemily committed:
Refactor text selectors and update app paths

- Simplify text selectors in group_tests.spec.ts by replacing "Contact mentions" with "Contact".
- Update app paths in capabilities_android.ts and capabilities_ios.ts to reflect new locations.
- Comment out debug output and unnecessary sleeps in utilities.ts for cleaner and more efficient code execution.

burtonemily committed:
Update README, streamline tests, and enhance utilities

- Updated README to reflect the new emulator specifications from Pixel 4 to Pixel 6 and system images from R to UpsideDownCake.
- Revised Android and iOS test scripts to use simplified video sending and handling disappearing messages more effectively.
- Enhanced environment variable setups to reference updated APK versions for Android.
- Removed outdated Appium setup instructions and introduced updated installation and test running options.
- Improved error handling in DeviceWrapper for StaleElementReferenceErrors and streamlined video sending functions.
- Adjusted capability setups to reflect current APK paths and added detailed debugging outputs.
- Updated utility functions for creating accounts and contacts to handle permissions more robustly.
- General cleanup of commented-out code blocks and updated documentation to match current test practices.

burtonemily committed:
Update tests and capabilities across multiple specs

- Updated link references in disappearing message checks and message checks specs for Android.
- Implemented platform-specific settings for disappearing message durations in Android and iOS tests.
- Enhanced user action sequences in Android user actions spec, improving reliability.
- Revised capabilities configuration for both Android and iOS, including path updates and added event timings.
- Modified control message strings for consistency and clarity in testing types.
- General cleanup and commented out unused code blocks, improving code readability and maintainability.

burtonemily committed:
Updated function clickOnElement to be clickOnByAccessibilityID for easier readability, updated disappearing messages tests to include dynamic mode and deleted control messages that are not correct

burtonemily committed:
Adding new test file

burtonemily committed:
Cleaning tests for iOS point release, updating accessibility ids and userflows that have changed and fixing coordinates test

burtonemily committed:
Merged dev into onboarding branch

burtonemily committed:
Merge branch 'main' into dev

burtonemily committed:
Cleaning up merge from dev to main

burtonemily committed:
Cleaning up merge

burtonemily committed:
Create contact update to include platform

burtonemily committed:
Updating test for android release, comments out control message checks until strings project, adds in sendVideo and sendDocument function, splits out some tests to separate ios and android functionality and adding in linked device testing for every test and removing the separate linked device testing

burtonemily committed:
Merge pull request #4 from burtonemily/android-fixes

Android fixes

burtonemily committed:
Adds in fixtures for basic test environments, create two accounts, link device, create contact and close all devices. Started updating user actions tests to include new fixture

Bilb commented on package-lock.json:
if you have a yarn.lock you shouldn't have a package-lock.json

yarn.lock is created by yarn and replaces package-lock.json (created by npm)

Bilb commented on appium_next.d.ts:
```diff
@@ -9,12 +9,14 @@ import { ExternalDriver } from "@appium/types/build/lib/driver";
 };
 
 export type AppiumNextElementType = {
+  value(arg0: string): unknown;
```
worth giving a better name to that arg0 maybe? Do we know what's its role?

burtonemily replied:
We're not using it anywhere, so I can't see the examples usage to know. Leave for now?

Bilb commented on package.json:
```diff
@@ -1,39 +1,47 @@
     "test": "_TESTING=1 mocha --timeout 300000 --retries 2 --jobs 1 ./run/test/specs/*.spec.js -- --grep ",
     "test-no-retry": "_TESTING=1 mocha --timeout 300000 --retries 0 --jobs 1 ./run/test/specs/*.spec.js -- --grep ",
-    "test-one": "_TESTING=1 mocha --timeout 300000 --retries 0 --jobs 1 ./run/test/specs/*.spec.js -- --grep ",
+    "test-one": "_TESTING=1  _FORCELOGS=1 mocha --timeout 300000 --retries 0 --jobs 1 ./run/test/specs/*.spec.js -- --grep",
```
nice

Bilb commented on run/constants/index.ts:
```diff
@@ -6,3 +6,32 @@ const ANDROID_XPATHS = {
+  READ: "read",
+};
+
+// const DISAPPEARING_CONTROL_MESSAGE = {
+//   YOU_LEGACY: `You set disappearing message time to 5 seconds`,
+//   LEGACY_DISAPPEARING_MESSAGE: `set disappearing message time to 5 seconds`,
+//   `${string} has set messages to disappear ${DISAPPEARING_TIMER} after they have been ${DISAPPEARING_ACTION}.`,
+//   : `${string} has set messages to disappear ${DISAPPEARING_TIMER} after they have been ${DISAPPEARING_ACTION}`,
+//   _3: `${string} has set messages to disappear ${DISAPPEARING_TIMER} after they have been ${DISAPPEARING_ACTION}`,
+//   _4: `${string} has set messages to disappear ${DISAPPEARING_TIMER} after they have been ${DISAPPEARING_ACTION}.`,
+//   _6: `${string} has set their messages to disappear ${DISAPPEARING_TIMER} after they have been ${DISAPPEARING_ACTION}.`;
+//   YOU: `You set your messages to disappear ${DISAPPEARING_TIMER} after they have been ${DISAPPEARING_ACTION}.`,
+// };
```
those should maybe be removed now that disappearing message is live? Or you might need to use some of those later?

burtonemily replied:
Removed

Bilb commented on run/test/specs/disappearing_messages.spec.ts:
```diff
@@ -1,72 +1,307 @@
-  androidIt("Disappearing messages", disappearingMessages);
+  // iosIt("Disappearing messages legacy", disappearingMessagesLegacy);
+  // androidIt("Disappearing messages legacy", disappearingMessagesLegacy);
+  iosIt("Disappear after send", disappearAfterSend);
+  androidIt("Disappear after send", disappearAfterSend);
+
+  iosIt("Disappear after read", disappearAfterRead);
+  androidIt("Disappear after read", disappearAfterRead);
+
+  iosIt("Disappear after send groups", disappearAfterSendGroups);
+  androidIt("Disappear after send groups", disappearAfterSendGroups);
+
+  iosIt("Disappear after send note to self", disappearAfterSendNoteToSelf);
+  androidIt("Disappear after send note to self", disappearAfterSendNoteToSelf);
```
for when you have the same name with the same function, you could create a mobileIt which would create a test for android and one for ios.
Does that sound useful? Do you need some clues of how to build it?

burtonemily replied:
Yeah that sounds good, so I created a mobileIt function 
```
export function mobileIt(
  platform: SupportedPlatformsType,
  title: string,
  test: (platform: SupportedPlatformsType) => Promise<void>
) {
  const testName = `${title} ${platform}`;

  return it(testName, async () => {
    console.info(`\n\n==========> Running "${testName}"\n\n`);
    await test(platform);
  });
}
```

and then I didn't want to specific each platform, and have the same issue so I created
```
export function bothPlatformsIt(
  title: string,
  test: (platform: SupportedPlatformsType) => Promise<void>
) {
  // Define test for Android
  mobileIt("android", title, test);

  // Define test for iOS
  mobileIt("ios", title, test);
}
```

Is this what you imagined?

Bilb commented on run/test/specs/disappearing_messages.spec.ts:
```diff
@@ -1,72 +1,307 @@
-  await device1.clickOnElement("Send message button");
-  // Wait 5 seconds
+  await device.disappearRadioButtonSelected("1 day");
+  await device.clickOnByAccessibilityID("10 seconds");
```
should come from `DISAPPEARING_TIMES` (also search and replace all of them matching it)

burtonemily replied:
So when I initially created the constants folder and used disappearing times as a constant, it turns out it makes it incredibly difficult to use the existing functions that look for or click on accessibility ID's with these constants. I"m going to have to create a new function that clicks on and looks for constants with this strict typing.. or if there's a better way you have in mind? (Will do some noodling on the idea)

Bilb commented on run/test/specs/disappearing_messages.spec.ts:
```diff
@@ -1,72 +1,307 @@
-  await device1.inputText("accessibility id", "Message input box", message);
-  await device1.clickOnElement("Send message button");
-  // Wait 5 seconds
+  await device.disappearRadioButtonSelected("1 day");
```
should come from DISAPPEARING_TIMES

Bilb commented on run/test/specs/disappearing_messages.spec.ts:
```diff
@@ -1,72 +1,307 @@
+  await runOnlyOnIOS(platform, () => device.navigateBack(platform));
+
+  await sleepFor(1000);
+  // await Promise.all([
+  //   device.waitForControlMessageToBePresent(
+  //     `${userA.userName} has set their messages to disappear 10 seconds after they have been sent.`
+  //   ),
+  //   device.waitForControlMessageToBePresent(
+  //     `${userA.userName} has set their messages to disappear 10 seconds after they have been sent.`
+  //   ),
+  // ]);
```
If those are eventually going to need to be enabled, you should add a warning to remember to enable them.
If not, can we remove them?

burtonemily replied:
Nah these need to be reenabled once the strings project is finished. What kind of warning were you thinking?

Bilb commented on run/test/specs/disappearing_messages.spec.ts:
```diff
@@ -1,72 +1,307 @@
+  // Change time to testing time of 10 seconds
+
+  if (platform === "android") {
+    time = "30 seconds";
```
DISAPPEARING_TIMES

Bilb replied:
not going to repeat this every where, but search for all the values from DISAPPEARING_TIMES which should be replaced

Bilb commented on run/test/specs/disappearing_messages.spec.ts:
```diff
@@ -1,72 +1,307 @@
+    newUser(device1, "Alice", platform),
+    newUser(device2, "Bob", platform),
+    newUser(device3, "Charlie", platform),
+  ]);
```
we will need to add fixtures for those eventually

Bilb commented on run/test/specs/disappearing_messages.spec.ts:
```diff
@@ -1,72 +1,307 @@
+  // Select disappearing messages option
   await sleepFor(1000);
-  await runOnlyOnAndroid(platform, () =>
+  if (platform === "android") {
```
rather than copy and pasting `platform === "android"` everywhere, you could create a 

isAndroid(platform) in utils somewher

Bilb replied:
actually you already have a isAndroid on the device itself, so you can do
if(device1.isAndroid()) {...}

feel free to replace all the other calls where === "android" or "ios" with the corresponding change

burtonemily replied:
would you prefer that to runOnlyOnAndroid()? isAndroid is a private function, i can export (make it public) it, if you prefer...?

burtonemily replied:
I've removed alot of the if(platform === 'blah') in most places and replaced it with runOnlyOnBlah

Bilb commented on run/test/specs/disappearing_messages.spec.ts:
```diff
@@ -1,72 +1,307 @@
+  await runOnlyOnIOS(platform, () => device1.navigateBack(platform));
+  // Check control message
+  await console.log(`Control message not working on Android: ignoring`);
+  // await Promise.all([
+  //   device1.disappearingControlMessage(
+  //     `You set messages to disappear ${time} after they have been ${action}.`
+  //   ),
+  //   device2.disappearingControlMessage(
+  //     `${userA.userName} has set messages to disappear ${time} after they have been ${action}.`
+  //   ),
+  //   device3.disappearingControlMessage(
+  //     `${userA.userName} has set messages to disappear ${time} after they have been ${action}.`
+  //   ),
+  // ]);
```
to keep or nah?

burtonemily replied:
To keep, just waiting on strings project

Bilb commented on run/test/specs/disappearing_messages.spec.ts:
```diff
@@ -1,72 +1,307 @@
+    device1.hasTextElementBeenDeleted("Message body", testMessage),
+    device2.hasTextElementBeenDeleted("Message body", testMessage),
+    device3.hasTextElementBeenDeleted("Message body", testMessage),
+  ]);
```
not mandatory change, but this might work too (actuaslly you might need a .bind here so let me know ifyou try to use this and have an issue 
```suggestion
  await Promise.all([device1, device2, device3].map(
     m => m.hasTextElementBeenDeleted("Message body", testMessage)
  ]);
```

Bilb commented on tsconfig.json:
```diff
@@ -5,7 +5,7 @@
     "removeComments": true,
     "noImplicitAny": true,
-    "skipLibCheck": true,
+    "skipLibCheck": false,
```
I might have done this change, but that's odd. Usually you want this to be true.

Bilb replied:
can you maybe turn it back to true and see if it breaks anything?

Bilb commented on run/types/testing.ts:
```diff
@@ -31,7 +85,30 @@ export type StrategyExtractionObj =
     };
 
+export type DisappearingControlMessage =
+  | `You set disappearing message time to 5 seconds`
+  | `${string} set disappearing message time to 5 seconds`
```
why not use the `DMTimeOption` type here to make this a type which can be anything  of
`You set disappearing message time to 5 seconds`
`You set disappearing message time to 10 seconds`
... 
automatically

burtonemily replied:
Updated

Bilb commented on run/types/DeviceWrapper.ts:
```diff
@@ -219,20 +236,22 @@ export class DeviceWrapper implements SharedDeviceInterface {
       x: xCoOrdinates,
       y: yCoOrdinates,
-      duration,
+      duration: duration,
```
no need to be explicit here

Bilb commented on run/test/specs/utils/utilities.ts:
```diff
@@ -61,24 +60,28 @@ export const installAppToDeviceName = async (
     `${adb} -s ${emulatorName} uninstall io.appium.settings`
   );
-  await sleepFor(100);
+  // await sleepFor(100);
```
a bunch of commented sleepFor (and others) can be removed here

Bilb commented on run/test/specs/utils/set_disappearing_messages.ts:
```diff
@@ -0,0 +1,49 @@
+  const enforcedType: ConversationType = conversationType;
+  await device.clickOnByAccessibilityID("More options");
+  await sleepFor(500);
+  if (platform === "ios") {
```
isIos(platform) ?

Bilb replied:
actually to make those, have a look at typescript type guards. Let me know if you don't understand why they are useful here

burtonemily replied:
Yeah I think you'll need to explain this to me :)

Bilb commented on run/test/specs/utils/open_app.ts:
```diff
@@ -213,6 +225,12 @@ const openiOSApp = async (
+  console.log(
+    `iOS App Full Path: ${
+      getIosCapabilities(capabilitiesIndex)["alwaysMatch"]["appium:app"]
+    }`
```
you could add chalk to make some of those console.log standout in the config  https://www.npmjs.com/package/chalk
(that's what I did on session-playwright)

Bilb commented on run/test/specs/utils/open_app.ts:
```diff
@@ -180,6 +189,11 @@ const openAndroidApp = async (
+    `Android App Full Path: ${
+      getAndroidCapabilities(capabilitiesIndex)["alwaysMatch"]["appium:app"]
+    }`
+  );
```
same comment with chalk

Bilb commented on run/test/specs/utils/open_app.ts:
```diff
@@ -165,8 +174,8 @@ const openAndroidApp = async (
   const emulatorAlreadyRunning = await isEmulatorRunning(targetName);
-  console.warn('emulatorAlreadyRunning', targetName, emulatorAlreadyRunning)
-  if(!emulatorAlreadyRunning) {
+  console.warn("emulatorAlreadyRunning", targetName, emulatorAlreadyRunning);
+  if (!emulatorAlreadyRunning) {
```
you need to setup prettier on vscode to avoid those kind of useless changes. check online or ask one of us if you can't make it work automaticailly on file being saved.

burtonemily replied:
I have Prettier setup, maybe its not working as intended...

Bilb commented on run/test/specs/utils/open_app.ts:
```diff
@@ -54,6 +59,10 @@ export const openAppTwoDevices = async (
+  function closeAllApps() {
+    // do the thing
+  }
+
```
was that a start of fixtures?

Bilb commented on run/test/specs/utils/join_community.ts:
```diff
@@ -0,0 +1,40 @@
+      strategy: "accessibility id",
+      selector: "Conversation header name",
+      text: communityName,
+    });
```
that part is shared for both platforms, so move it out of the else at the end of the function instead.

Bilb commented on run/test/specs/utils/create_contact.ts:
```diff
@@ -15,22 +15,44 @@ export const newContact = async (
+    console.log(`Retrying message request`);
+  } else {
+    console.log("Found message request banner: No need for retry");
+  }
```
wouldn't this throw if the element is not found?

Bilb commented on run/test/specs/utils/create_account.ts:
```diff
@@ -25,28 +25,37 @@ export const newUser = async (
 
   console.log(`${userName}s sessionID found: "${sessionID}" "${platform}"`);
-
+  // TODO add 'allow notifications' function here for android API 34
```
you can probably know which target is a device via ADB, so you could make this work on both, depending of > 24 or not

Bilb commented on run/test/specs/utils/click_by_coordinates.ts:
```diff
@@ -1,10 +1,14 @@
 ) => {
-  await device.pressCoordinates(xCoOrdinates, yCoOrdinates);
-  console.log(`Tapped coordinates ${xCoOrdinates}, ${yCoOrdinates}`);
+  const { x, y } = coordinates;
```
could be done in the argument above directly


```ts
export const clickOnCoordinates = async (
  device: DeviceWrapper,
  {x, y}: Coordinates
```

Bilb commented on package.json:
```diff
@@ -1,39 +1,47 @@
   "devDependencies": {
     "@appium/execute-driver-plugin": "^3.0.1",
     "@appium/images-plugin": "^2.0.1",
+    "@types/jest": "^29.2.5",
```
do we need jest? and @types/jest. It seems that we only use mocha to run the tests (see the scripts section above)

burtonemily replied:
Yeah i'm not sure why jest is here, i'll try uninstalling and hope it doesn't break everything 😅

burtonemily replied:
Removed jest and added missing types for mocha

Bilb commented on package.json:
```diff
@@ -1,39 +1,47 @@
     "@wdio/types": "^8.1.2",
-    "appium-uiautomator2-driver": "^2.12.1",
-    "appium-xcuitest-driver": "^4.16.7",
+    "ci": "^2.2.0",
```
this might have been added by mistake no? 
Or do we really want this package? https://www.npmjs.com/package/ci and do we use it?

burtonemily replied:
Ummmm I don't believe I installed this on purpose, so I'll uninstall

Bilb commented on run/test/specs/call_checks.spec.ts:
```diff
@@ -118,53 +124,81 @@ async function voiceCallIos(platform: SupportedPlatformsType) {
-  await device2.clickOnElement("Settings");
+  // Need to navigate out of conversation for user to have full contact actions (calls icon, etc)
+  await device2.navigateBack(platform);
+  await device2.clickOnElementAll({
```
should you not merge those two functions together?

```ts
  public async clickOnElementAll(
    args: { text?: string; maxWait?: number } & StrategyExtractionObj
  ) {
    let el: null | AppiumNextElementType = null;
    const { text } = args;
    if (text) {
      el = await this.waitForTextElementToBePresent({ ...args, text });
    } else {
      el = await this.waitForTextElementToBePresent(args);
    }
    await this.click(el.ELEMENT);
    return el;
  }

  public async clickOnElementByText(
    args: { text: string; maxWait?: number } & StrategyExtractionObj
  ) {
    const { text } = args;
    const el = await this.waitForTextElementToBePresent(args);

    if (!el) {
      throw new Error(`clickOnElementByText: Couldnt find text: ${text}`);
    }
    await this.click(el.ELEMENT);
  }
```

Bilb commented on run/test/specs/call_checks.spec.ts:
```diff
@@ -82,28 +81,35 @@ async function voiceCallAndroid(platform: SupportedPlatformsType) {
-  await device2.findConfigurationMessage(`${userA.userName} called you`);
+  await device1.waitForControlMessageToBePresent(`Called ${userB.userName}`);
+  await device2.waitForControlMessageToBePresent(
+    `${userA.userName} called you`
```
this is nice

Bilb commented on run/test/specs/community_tests.spec.ts:
```diff
@@ -0,0 +1,139 @@
+
+async function sendImageCommunityiOS(platform: SupportedPlatformsType) {
+  const { device1, device2 } = await openAppTwoDevices(platform);
+  const testCommunityLink = `https://chat.lokinet.dev/testing-all-the-things?public_key=1d7e7f92b1ed3643855c98ecac02fc7274033a3467653f047d6e433540c03f17`;
```
this should maybe be a constant somewhere and not be copy/pasted

Bilb commented on run/test/specs/community_tests.spec.ts:
```diff
@@ -0,0 +1,139 @@
+  // await Promise.all([
+  //   device1.scrollToBottom(platform),
+  //   device2.scrollToBottom(platform),
+  // ]);
```
needed? (a few of those in the file)

Bilb commented on run/test/specs/media/test_file.pdf:
this file should be somewhere else than here. Probably in a folder in the root directly

i.e. at the same level as run

burtonemily replied:
This must an old commit, it has already been moved :)

Bilb commented on run/types/testing.ts:
```diff
@@ -31,7 +85,30 @@ export type StrategyExtractionObj =
+  | `Title is now '${string}'.`
+  | `You renamed the group to ${string}`
+  | `${string} joined the group.`
+  | `You added ${string} to the group.`;
```
any type with ${string} in it is useless. I think we will have to debug this together to understand why you've needed to add them and not a subset of string

Bilb replied:
come grab me when you want to have a look at it.

Bilb replied:
you probably just need a bunch of `as const`

Bilb commented on run/test/specs/group_tests.spec.ts:
```diff
@@ -309,45 +263,23 @@ async function mentionsForGroups(platform: SupportedPlatformsType) {
-  // Check in User C's device if the format is correct
-  await device3.findMessageWithBody("@You");
+  // Bob to Select User C
+  await device2.tagContact(platform, userC);
```
mentionContact rather than tagContact?

burtonemily replied:
haha ok, i'll change

Bilb commented on run/test/specs/linked_device_group.spec.ts:
```diff
@@ -20,97 +20,155 @@ async function groupCreationandNameChangeLinkedDevice(
+    device2.waitForControlMessageToBePresent(`Title is now '${newGroupName}'.`),
+    device3.waitForControlMessageToBePresent(`Title is now '${newGroupName}'.`),
+    device4.waitForControlMessageToBePresent(`Title is now '${newGroupName}'.`),
+  ]);
```
you could make a utility function running a task for each devices in an array of devices (as you seem to be doing this quite often). Making it typesafe is going to be fun, but would be a great addition

i.e.
```ts
await onEachDevice([device2, device3, device4], 
  'waitForControlMessageToBePresent', `Title is now '${newGroupName}'.`)
```

grab me for help to write it

Bilb commented on run/test/specs/linked_device_group.spec.ts:
```diff
@@ -179,3 +250,80 @@ describe("Linked device - group tests", () => {
+//   const testGroupName = "Disappearing messages group";
+//   const { device1, device2, device3, device4 } = await openAppFourDevices(
+//     platform
+//   );
```
needs to be kept?

Bilb commented on run/test/specs/linked_device_user.spec.ts:
```diff
@@ -233,27 +224,27 @@ async function avatarRestorediOS(platform: SupportedPlatformsType) {
   const profilePicture = await device1.doesElementExist({
     strategy: "accessibility id",
-    selector: `Photo, May 01, 1998, 7:00 AM`,
+    selector: `Photo, 01 May 1998, 7:00 am`,
```
is this not a localization of the emulator issue?

burtonemily replied:
This seems to change with each OS update, not sure how to fix this

Bilb commented on run/test/specs/linked_device_user.spec.ts:
```diff
@@ -271,73 +262,71 @@ async function avatarRestorediOS(platform: SupportedPlatformsType) {
+  await sleepFor(500);
+  await device1.clickOnElementAll({
+    strategy: "xpath",
+    selector: `/hierarchy/android.widget.FrameLayout/android.widget.LinearLayout/android.widget.FrameLayout/android.widget.ScrollView/android.widget.TabHost/android.widget.LinearLayout/android.widget.FrameLayout/androidx.viewpager.widget.ViewPager/android.widget.RelativeLayout/android.widget.GridView/android.widget.LinearLayout/android.widget.LinearLayout[2]`,
```
should this selector be extracted to the constant list of xpaths?

Bilb replied:
in testing.ts

burtonemily replied:
I had started to extract out to the constants -> index.ts, but never finished. Will update this now

Bilb commented on run/test/specs/linked_device_user.spec.ts:
```diff
@@ -350,43 +339,53 @@ async function avatarRestoredAndroid(platform: SupportedPlatformsType) {
+    });
+    await device1.clickOnElementAll({
+      strategy: "xpath",
+      selector: `/hierarchy/android.widget.FrameLayout/android.widget.FrameLayout/android.widget.ListView/android.widget.LinearLayout`,
```
should be moved to testing.ts too

Bilb commented on run/test/specs/linked_device_user.spec.ts:
```diff
@@ -350,43 +339,53 @@ async function avatarRestoredAndroid(platform: SupportedPlatformsType) {
   const pixelColor = await parseDataImage(base64);
   console.log("RGB Value of pixel is:", pixelColor);
-  if (pixelColor === "03cbfe") {
+  if (pixelColor === "cbfeff") {
```
why this this change?

Bilb commented on run/test/specs/linked_device_user.spec.ts:
```diff
@@ -350,43 +339,53 @@ async function avatarRestoredAndroid(platform: SupportedPlatformsType) {
     console.log("Colour is correct on device 1");
   } else {
-    console.log("Colour isn't 03cbfe, it is: ", pixelColor);
+    console.log("Colour isn't cbfeff, it is: ", pixelColor);
```
probably worth creating a sneaky variable so you don't have to update two places each time the color changes

burtonemily replied:
Great idea, will do

Bilb commented on run/test/specs/linked_device_user.spec.ts:
```diff
@@ -350,43 +339,53 @@ async function avatarRestoredAndroid(platform: SupportedPlatformsType) {
   const pixelColorLinked = await parseDataImage(base64A);
-  if (pixelColorLinked === "03cbfe") {
+  if (pixelColorLinked === "cbfeff") {
     console.log("Colour is correct on linked device");
   } else {
-    console.log("Colour isn't 03cbfe, it is: ", pixelColorLinked);
+    console.log("Colour isn't cbfeff, it is: ", pixelColorLinked);
```
actually you definitely need a variable

burtonemily replied:
I don't even know what the original colour was anymore 😅

Bilb commented on run/test/specs/utils/capabilities_android.ts:
```diff
@@ -1,23 +1,21 @@
 import { CapabilitiesIndexType } from "./capabilities_ios";
 
-const androidAppFullPath = `${getAndroidBinariesRoot()}/session-1.17.0-universal.apk`;
+const androidAppFullPath = `/Users/emilyburton/Documents/session-android/app/build/outputs/apk/play/debug/session-1.18.2-universal.apk`;
```
why did you remove the getAndroidBinariesRoot?

Bilb replied:
this was made to avoid commiting paths that only you can use.

burtonemily replied:
Ah I forgot about that, I'm constantly changing the path so it was just an oversight, will update to use binaries root

Bilb commented on run/test/specs/utils/capabilities_android.ts:
```diff
@@ -1,23 +1,21 @@
     "org.thoughtcrime.securesms.onboarding.LandingActivity",
   "appium:automationName": "UiAutomator2",
   "appium:newCommandTimeout": 300000,
+  "appium:eventTimings": true,
```
this might make the tests a lot slower. Should we have disabled by default, but turn it on when needed?

Bilb commented on run/types/DeviceWrapper.ts:
```diff
@@ -219,20 +236,22 @@ export class DeviceWrapper implements SharedDeviceInterface {
+      duration: duration,
     };
+
+    const actions = [action];
```
no need for a temporary variable, just call performTouch with ([action]) below

Bilb commented on run/types/DeviceWrapper.ts:
```diff
@@ -562,6 +601,26 @@ export class DeviceWrapper implements SharedDeviceInterface {
+  // ): Promise<AppiumNextElementType> {
+  //   let el: null | AppiumNextElementType = null;
+  //   console.log(
+  //     `Waiting for accessibility ID '${selector}' to be present with ${text}`
```
can be removed?

Bilb commented on run/types/DeviceWrapper.ts:
```diff
@@ -705,6 +747,46 @@ export class DeviceWrapper implements SharedDeviceInterface {
     return element;
   }
 
+  public async hasElementBeenDeletedNew({
```
should probably rename that function to a real name now

burtonemily replied:
updated :)

Bilb commented on run/types/DeviceWrapper.ts:
```diff
@@ -705,6 +747,46 @@ export class DeviceWrapper implements SharedDeviceInterface {
+          console.log(`Text element has been deleted, great success`);
+        }
+      }
+    } while (Date.now() - start <= maxWait && element);
```
there is no need to separate those two calls in the if/else as  waitForTextElementToBePresent allows an argument text unset. Let me klnow if you have issues but you should be able to just 

```ts
 do {
        try {
          element = await this.waitForTextElementToBePresent({
            text,
            maxWait: 100,
            ...args,
          });
          await sleepFor(100);
          console.log(`Element has been found, waiting for deletion`);
        } catch (e) {
          element = undefined;
          console.log(`Element has been deleted, great success`);
        }
    } while (Date.now() - start <= maxWait && element);
```

Bilb commented on run/types/DeviceWrapper.ts:
```diff
@@ -705,6 +747,46 @@ export class DeviceWrapper implements SharedDeviceInterface {
+          await sleepFor(100);
+          console.log(`Text element has been found, waiting for deletion`);
+        } catch (e) {
+          element = undefined;
```
no need to explicitely set element to undefined as it would already be undefined if waitForTextElementToBePresent did throw

Bilb commented on run/types/DeviceWrapper.ts:
```diff
@@ -736,87 +818,149 @@ export class DeviceWrapper implements SharedDeviceInterface {
     let currentWait = 0;
     const waitPerLoop = 100;
-    let el: AppiumNextElementType | null = null;
+    while (el === null) {
```
this should probably be while (!el) to take care of the undefined case too

Bilb commented on run/types/DeviceWrapper.ts:
```diff
@@ -933,12 +1080,288 @@ export class DeviceWrapper implements SharedDeviceInterface {
+        console.log("Great success - default time is correct");
+      } else {
+        throw new Error("Dammit - default time was not correct");
+      }
```
```suggestion
      if (attr === "selected") {
        console.log("Great success - default time is correct");
        return;
      }        
      throw new Error("Dammit - default time was not correct");
      
```

Bilb commented on run/types/DeviceWrapper.ts:
```diff
@@ -933,12 +1080,288 @@ export class DeviceWrapper implements SharedDeviceInterface {
+    message?: string,
+    community?: boolean
+  ) {
+    if (platform === "ios") {
```
```suggestion
if(this.isIos())
```

same for other calls

Bilb commented on run/types/DeviceWrapper.ts:
```diff
@@ -933,12 +1080,288 @@ export class DeviceWrapper implements SharedDeviceInterface {
+    await sleepFor(100);
+    await this.clickOnByAccessibilityID("Images folder");
+    await this.clickOnByAccessibilityID("Continue");
+    await this.clickOnElementAll({
```
I would say that those sendImageWith and sendImage() should not be part of the deviceWrapper class, as they are pretty high level

Bilb replied:
same for video

Bilb replied:
I just feel like the device wrapper is going to get out of hand otherwise.

Bilb commented on run/types/DeviceWrapper.ts:
```diff
@@ -980,11 +1403,25 @@ export class DeviceWrapper implements SharedDeviceInterface {
+  }
+
   public async navigateBack(platform: SupportedPlatformsType) {
     if (platform === "ios") {
```
this.isIos() You don't need the argument to navigateBack

Bilb commented on run/test/specs/message_checks_android.spec.ts:
```diff
@@ -292,18 +232,18 @@ async function sendLongMessage(platform: SupportedPlatformsType) {
 async function sendLink(platform: SupportedPlatformsType) {
   const { device1, device2 } = await openAppTwoDevices(platform);
-
+  const testLink = `https://type-level-typescript.com/objects-and-records`;
```
I wouldn't use real urls for this. What about example.org? or getsession.org if really.

Bilb commented on run/test/specs/message_checks_android.spec.ts:
```diff
@@ -312,36 +252,57 @@ async function sendLink(platform: SupportedPlatformsType) {
+
+async function sendCommunityInviteMessage(platform: SupportedPlatformsType) {
+  const { device1, device2 } = await openAppTwoDevices(platform);
+  const communityLink = `https://chat.lokinet.dev/testing-all-the-things?public_key=1d7e7f92b1ed3643855c98ecac02fc7274033a3467653f047d6e433540c03f17`;
```
to a constant

Bilb commented on run/test/specs/message_checks_groups_android.spec.ts:
```diff
@@ -0,0 +1,479 @@
+  await device1.longPress("New voice message");
+  await device1.clickOnByAccessibilityID("Continue");
+  await device1.clickOnElementXPath(
+    `/hierarchy/android.widget.FrameLayout/android.widget.FrameLayout/android.widget.FrameLayout/android.widget.ScrollView/android.widget.LinearLayout/android.widget.LinearLayout/android.widget.LinearLayout[2]/android.widget.Button[1]`
```
to the testing.ts

Bilb commented on run/test/specs/message_checks_groups_android.spec.ts:
```diff
@@ -0,0 +1,479 @@
+  // Create contact between User A and User B
+  await createGroup(
+    platform,
+    device1,
```
there is a lot of duplicate setup for those attachments tests

Bilb replied:
even without fixture, you could create a function in the file to do the setup for you and call it at the start of each tests, rather than having so much copy and paste

Bilb commented on run/test/specs/message_checks_groups_android.spec.ts:
```diff
@@ -0,0 +1,479 @@
+    userC,
+    testGroupName
+  );
+  const testLink = `https://example.org/`;
```
:+1:

Bilb commented on run/test/specs/message_checks_groups_android.spec.ts:
```diff
@@ -0,0 +1,479 @@
+  // Select gif
+  await device1.clickOnElementAll({
+    strategy: "xpath",
+    selector: `/hierarchy/android.widget.FrameLayout/android.widget.LinearLayout/android.widget.FrameLayout/android.widget.FrameLayout/android.widget.FrameLayout/android.widget.RelativeLayout/android.widget.ScrollView/androidx.viewpager.widget.ViewPager/android.widget.FrameLayout/androidx.recyclerview.widget.RecyclerView/android.widget.FrameLayout[1]`,
```
to testing.ts

Bilb commented on run/test/specs/message_checks_groups_android.spec.ts:
```diff
@@ -0,0 +1,479 @@
+    testGroupName
+  );
+  const longText =
+    "Mauris sapien dui, sagittis et fringilla eget, tincidunt vel mauris. Mauris bibendum quis ipsum ac pulvinar. Integer semper elit vitae placerat efficitur. Quisque blandit scelerisque orci, a fringilla dui. In a sollicitudin tortor. Vivamus consequat sollicitudin felis, nec pretium dolor bibendum sit amet. Integer non congue risus, id imperdiet diam. Proin elementum enim at felis commodo semp ...
```
should maybe be a constant too?

Bilb commented on run/test/specs/message_checks_groups_ios.spec.ts:
```diff
@@ -0,0 +1,516 @@
+      } 'run/test/specs/media/test_image.jpg'`,
+      true
+    );
+  }
```
hasn't this logic been moved to devicewrapper now?

Bilb commented on run/test/specs/message_checks_groups_ios.spec.ts:
```diff
@@ -0,0 +1,516 @@
+      `touch -a -m -t ${bestDayOfYear} 'run/test/specs/media/test_video.mp4'`,
+      true
+    );
+    await runScriptAndLog(
```
same question

Bilb commented on run/test/specs/message_checks_groups_ios.spec.ts:
```diff
@@ -0,0 +1,516 @@
+    device3,
+    userC,
+    testGroupName
+  );
```
lots of duplicates again between those group tests

Bilb commented on run/test/specs/message_checks_groups_ios.spec.ts:
```diff
@@ -0,0 +1,516 @@
+async function sendLongMessageGroup(platform: SupportedPlatformsType) {
+  const testGroupName = "Message checks for groups";
+  const longText =
+    "Mauris sapien dui, sagittis et fringilla eget, tincidunt vel mauris. Mauris bibendum quis ipsum ac pulvinar. Integer semper elit vitae placerat efficitur. Quisque blandit scelerisque orci, a fringilla dui. In a sollicitudin tortor. Vivamus consequat sollicitudin felis, nec pretium dolor bibendum sit amet. Integer non congue risus, id imperdiet diam. Proin elementum enim at felis commodo semp ...
```
to constants?

Bilb commented on run/test/specs/message_checks_ios.spec.ts:
```diff
@@ -366,7 +384,7 @@ async function sendLongMessage(platform: SupportedPlatformsType) {
 async function sendLink(platform: SupportedPlatformsType) {
   const { device1, device2 } = await openAppTwoDevices(platform);
-
+  const testLink = `https://type-level-typescript.com/objects-and-records`;
```
example.org or getsession.org instead?

Bilb commented on run/test/specs/message_checks_ios.spec.ts:
```diff
@@ -375,39 +393,88 @@ async function sendLink(platform: SupportedPlatformsType) {
+
+async function sendCommunityInvitation(platform: SupportedPlatformsType) {
+  const { device1, device2 } = await openAppTwoDevices(platform);
+  const communityLink = `https://chat.lokinet.dev/testing-all-the-things?public_key=1d7e7f92b1ed3643855c98ecac02fc7274033a3467653f047d6e433540c03f17`;
```
to testing.ts or as a constant

Bilb commented on run/test/specs/message_checks_ios.spec.ts:
```diff
@@ -375,39 +393,88 @@ async function sendLink(platform: SupportedPlatformsType) {
+  //   strategy: "accessibility id",
+  //   selector: "Username",
+  //   text: communityName,
+  // });
```
needed?

Bilb commented on run/test/specs/user_actions.spec.ts:
```diff
@@ -37,33 +44,33 @@ async function blockUserInConversationOptions(
-
   // On ios, you need to navigate back to conversation screen to confirm block
-  await runOnlyOnIOS(platform, () => device1.clickOnElement("Back"));
+  await runOnlyOnIOS(platform, () => device1.navigateBack(platform));
```
after our discussion, we could make the runOnlyOnIOS/android  a bit cleaner by adding to the class DeviceWrapper a public method as so:

```ts
  public onIOS(): Omit<DeviceWrapper, "runOnlyOnIOS" | "runOnlyOnAndroid"> {
    if (this.isIOS()) {
      return omit(this, "runOnlyOnIOS", "runOnlyOnAndroid"); // needed to avoid circular reference to that same method
    }
    return stubbed;
  }
```

where `stubbed` is
`const stubbed = Sinon.createStubInstance(DeviceWrapper);`
and with
```bash
yarn add sinon
yarn add -D @types/sinon
```

For details check our chat on #session-all-platforms of the 27/05/2024

Bilb replied:
this would allow us to do
`device.runOnlyOnIOS().navigateBack()`

Bilb replied:
another comment on the same way would be to make the calls on DeviceWrapper which do not return anything currently `return this` so we could chain them.
For instance `device.runOnlyOnIOS().clickOn("test1").clickOn("test2")`

Bilb commented on run/test/specs/user_actions.spec.ts:
```diff
@@ -135,47 +142,58 @@ async function changeUsername(platform: SupportedPlatformsType) {
   await closeApp(device);
 }
-
+// TO FIX (WRONG USER FLOW?)
```
not sure what those to fix all over the place are for, but they should be fixed I guess before merge

Bilb commented on run/test/specs/user_actions.spec.ts:
```diff
@@ -188,26 +206,26 @@ async function changeProfilePictureAndroid(platform: SupportedPlatformsType) {
   const pixelColor = await parseDataImage(base64);
   console.log("RGB Value of pixel is:", pixelColor);
-  if (pixelColor === "03cbfe") {
+  if (pixelColor === "cbfeff") {
```
make this a const (in the function or shared as you have it in a few files) and reuse it.

Bilb commented on run/test/specs/disappearing_messages.spec.ts:
```diff
@@ -1,72 +1,307 @@
+    time = "30 seconds";
+    await device1.clickOnByAccessibilityID(time);
+  } else {
+    time = "10 seconds";
```
same

Bilb commented on run/test/specs/user_actions.spec.ts:
```diff
@@ -283,64 +302,67 @@ async function setNicknameAndroid(platform: SupportedPlatformsType) {
-  const revertedHeader = await device1.waitForElementToBePresent({
+  await device1.selectByText("Conversation list item", nickName);
+  const changedListName = await device1.waitForTextElementToBePresent({
     strategy: "accessibility id",
-    selector: "Username",
+    selector: "Conversation header name",
+    text: nickName,
```
naming isn't great here, changedListName, listName, and nickname.
it's ok  for the last 2, but changedListName is actually an element, so maybe listNameElement instead?

Bilb commented on run/test/specs/user_actions.spec.ts:
```diff
@@ -377,23 +406,24 @@ async function setNicknameIos(platform: SupportedPlatformsType) {
-  await device1.clickOnElement("Done");
+  await device1.deleteText("Username");
+  await device1.inputText("accessibility id", "Username", " ");
+  await await device1.clickOnByAccessibilityID("Done");
```
search and replace all double await, this is a typo and undeeded

Bilb commented on run/test/specs/disappearing_message_checks_android.spec.ts:
```diff
@@ -0,0 +1,477 @@
+    ["1:1", "Disappear after send option", "1 minute"],
+    device2
+  );
+  // TODO FIX CONTROL MESSAGES ON ANDROID
```
you should really have a way to have those somewhere in the report. so you have a summary of what's left.
Maybe skip the test entirely in the androidIt call? (this needs to  be built)

Bilb commented on run/test/specs/disappearing_message_checks_android.spec.ts:
```diff
@@ -0,0 +1,477 @@
+  await setDisappearingMessage(
+    platform,
+    device1,
+    ["1:1", "Disappear after send option", "1 minute"],
```
should this be reusing the var time/mode?

burtonemily replied:
yes.... technically
I'll update the time, i'm just struggling with the mode, there's alot of annoying nuances to 'send'/'sent' or 'read'. Will explain in person haha

Bilb commented on run/test/specs/disappearing_message_checks_android.spec.ts:
```diff
@@ -0,0 +1,477 @@
+    newUser(device1, "Alice", platform),
+    newUser(device2, "Bob", platform),
+  ]);
+  const time: DMTimeOption = "1 minute";
+  const mode: DisappearActions = "sent";
```
`const time: DMTimeOption = "1 minute";` works, but my prefered way for those (unless if a let and not a const) is actually to do 
`const time = "1 minute" as const;`
this makes `time` type be a string which can only be "1 minute"
So when you try to call a function which takes an argument of type `DMTimeOption` which is the same as `"1 minute" | "5 minutes" ...` typescript is happy

ask me for details if needed

Bilb replied:
this comment applies to the whole file (and maybe others, search and replace)

Bilb commented on run/test/specs/disappearing_message_checks_android.spec.ts:
```diff
@@ -0,0 +1,477 @@
+  await device1.longPress("New voice message");
+  await device1.clickOnByAccessibilityID("Continue");
+  await device1.clickOnElementXPath(
+    `/hierarchy/android.widget.FrameLayout/android.widget.FrameLayout/android.widget.FrameLayout/android.widget.ScrollView/android.widget.LinearLayout/android.widget.LinearLayout/android.widget.LinearLayout[2]/android.widget.Button[1]`
```
to testing.ts

Bilb commented on run/test/specs/disappearing_message_checks_android.spec.ts:
```diff
@@ -0,0 +1,477 @@
+  });
+  // Select gif
+  await sleepFor(500);
+  await device1.clickOnElementXPath(
```
you can search for that function an make sure the xpath is part of a constant in testing.ts (where other xpaths are defined)

Bilb commented on run/test/specs/disappearing_message_checks_android.spec.ts:
```diff
@@ -0,0 +1,477 @@
+  platform: SupportedPlatformsType
+) {
+  const { device1, device2 } = await openAppTwoDevices(platform);
+  const communityLink = `https://chat.lokinet.dev/testing-all-the-things?public_key=1d7e7f92b1ed3643855c98ecac02fc7274033a3467653f047d6e433540c03f17`;
```
from constants.ts/testing.ts, search and replace to make sure there are no left over of this exept in one file

burtonemily committed:
Removes jest and installs mocha types dependency to replace jest

burtonemily committed:
Increases sleepFor to account for changes on android

burtonemily committed:
Enhance community interaction and update utility defaults

- Modify reply message format in community_tests.spec.ts to include a timestamp, enhancing traceability of interactions.
- Update the accessibility ID in join_community.ts from "Join community option" to "Join community button" to reflect UI changes and improve code clarity.
- Increase the default maximum wait time in DeviceWrapper from 15 seconds to 60 seconds, improving reliability under slower network conditions or heavier load.

burtonemily committed:
Addresses feedback

burtonemily committed:
Addresses feedback

burtonemily committed:
Merge pull request #5 from burtonemily/android-fixes

Android fixes

burtonemily committed:
Merge branch 'main' into onboarding

burtonemily committed:
Merges main into onboarding

burtonemily committed:
Changes disappearing messages time to use constants and tries to add strategy DMTimeOption

burtonemily committed:
Skipping flaky tests

burtonemily committed:
Implements (and finishes) constants for all disappearing messages timers

burtonemily committed:
Finishes off addressing feedback

burtonemily committed:
Merge branch 'main' into addressing-feedback

burtonemily committed:
Merge pull request #6 from burtonemily/addressing-feedback

burtonemily committed:
Starts adding in before function

burtonemily committed:
Combines add-linked-device branch into this branch because there was a function started that I wanted to utilise in this branch

burtonemily committed:
Starts abstracting test set up into before block to reduce test times

burtonemily committed:
Removes test setup into own function and into before for message checks 1:1 on ios

burtonemily committed:
Adds in a script that isolate describe blocks

burtonemily committed:
- Updated `package.json` scripts:
  - Merged iOS and Android specific scripts to simplify usage.
  - Improved `test-describe-ios` and `test-describe-android` scripts to handle GREP patterns properly.
- Refactored multiple test files to split iOS and Android testsAdds in a script that isolate describe blocks.

burtonemily committed:
Starts parallelising tests into groups, still a WIP

burtonemily committed:
Extends timer to 1 minute from 10 seconds and switches DaS options in 1:1 tests to DaR while libsession is in testing phase

burtonemily committed:
Reverts changes on disappearing messages to use new function

burtonemily committed:
Updates the tests to use 1 minute and DaR instead of 10 seconds on DaS

burtonemily committed:
Removes flaky send image to community test and updates sleepFor

burtonemily committed:
Merges main into reducing test time

burtonemily committed:
Updates ios capabilities to include mochas worker id and splits out emulators into two distinct groups for parallel testing

burtonemily committed:
Updates disappearing mesages in groups to include new test times for ios and includes a new test file for parallel testing

burtonemily committed:
Updates a few dependencies

burtonemily committed:
Updating file structure, so tests are in their own file instead of grouped with describe blocks, to make accommodations for parallel testing

burtonemily committed:
Updates first simulator issue when adding media to a device, also updates sending video workflow into its own function

burtonemily committed:
Updates some discrepancies from latest android 1.18.5 release

burtonemily committed:
Adds in conversation type to sendImage function to account for keyboard being open or closed dependent on type - Adds in 30 second timer for disappearing messages for ios to account for sending times extending -Changes addMedia function to device to account for parallel testing

burtonemily committed:
Adds in a default timer for disappearing messages for both platforms as 30 seconds and removes setting the timer in the test individually

burtonemily committed:
Addresses feedback

burtonemily committed:
Merge remote-tracking branch 'origin/main' into reducing_test_time

burtonemily committed:
Merge pull request #7 from burtonemily/reducing_test_time

burtonemily committed:
Removes duplicate files from js and attempts to implement playwright instead of mocha for parallel testing

burtonemily committed:
Updates test tags to reflect current changes in onboarding

burtonemily committed:
Updates parallel execution to include android emulators

burtonemily committed:
Updates gitignore

burtonemily committed:
Updates gitignore

burtonemily committed:
removed .yarnrc.yml file

burtonemily committed:
Merge pull request #8 from burtonemily/reducing_test_time

Bilb committed:
adding yarn cjs because they are a pain

Bilb committed:
bump a bunch of deps for appium

burtonemily committed:
Merge pull request #10 from Bilb/fix-build

burtonemily committed:
Updates package.json file

burtonemily committed:
bump appium image plugin version

burtonemily committed:
make patch ofr appium support logger mocked log

burtonemily committed:
Updates tests to match onboarding changes

burtonemily committed:
Adds env file into gitignore

burtonemily committed:
Merge remote-tracking branch 'origin/main' into onboarding

burtonemily committed:
Removes mocha and finishes dealing with merge conflicts

burtonemily committed:
Updates some discrepancies between android and ios test tags

burtonemily committed:
Starts using Locators to simplify locating elements for both platforms

burtonemily committed:
Starts using Locators to simplify locating elements for both platforms

burtonemily committed:
Updates yarn

burtonemily committed:
Merge branch 'locators_WIP' of github.com:burtonemily/session-appium into locators_WIP

burtonemily committed:
Tracks the locator folder

burtonemily committed:
Tracks the locator folder

burtonemily committed:
Merge branch 'locators_WIP' of github.com:burtonemily/session-appium into locators_WIP

burtonemily committed:
Implementing new locators throughout tests

burtonemily committed:
Merge branch 'onboarding'

burtonemily committed:
Merge branch 'main' into locators_WIP

burtonemily committed:
remove appium fake types and use node_modules ones

burtonemily committed:
remove appium fake types and use node_modules ones

burtonemily committed:
Revert "remove appium fake types and use node_modules ones"

This reverts commit 55b5f032761db0969ecd08ef84014b540d8c8feb.

burtonemily committed:
Removes chalk

burtonemily committed:
Merging changes

burtonemily committed:
Updates new fix for XCUITest modal removal for active app

burtonemily committed:
Changes some test tags for new Onboarding changes, updates test to account for breaking changes on android new release, updates session reporter to include skipped tests and removes multiple workers for android testing

burtonemily committed:
Adds prettier config

burtonemily committed:
Updates prettier and therefore some formatting changes

burtonemily committed:
Adding in playwright config file

burtonemily committed:
Addresses feedback

burtonemily committed:
Adds prettier ignore and updates change username to include new locator

burtonemily committed:
Merge branch 'main' into locators_WIP

burtonemily committed:
Limits timeouts on modal permission

burtonemily committed:
Merge branch 'locators_WIP' of github.com:burtonemily/session-appium into locators_WIP

burtonemily committed:
Addresses feedback and includes a bunch of prettier formatting...

burtonemily committed:
Chore: run prettier on all files

burtonemily committed:
Addresses feedback

burtonemily committed:
Inputs new modal dismissal functionality and adds in a retry check for app launch failure

burtonemily committed:
Refactoring tests to account for Appium update, changed console warn -> console info and updates some locator test tags for new onboarding changes

burtonemily committed:
Addresses feedback

burtonemily committed:
Addresses feedback

burtonemily committed:
Addresses feedback

burtonemily committed:
Merge pull request #12 from burtonemily/locators_WIP

Locators wip
